tests/cases/compress_request_body.rs

```diff
@@ -0,0 +1,209 @@
+
+#[test]
+fn skip_compression_when_compression_ratio_is_negative() {
+    let server = server();
+    get_command()
+        .arg(format!("{}/normal", server.base_url()))
+        .args([&format!("key={}", "1"), "-x", "-f", "--pretty=none"])
+        .assert()
+        .stdout(indoc::formatdoc! {r#"
+            HTTP/1.1 200 OK
+            Date: N/A
+            Content-Type: text/plain
+            Content-Length: 5
+
+            key={c}
+        "#, c = "1"});
+}
+
+#[test]
+fn test_compress_force_with_negative_ratio() {
+    let server = server();
+    get_command()
+        .arg(format!("{}/deflate", server.base_url()))
+        .args([&format!("key={}", "1"), "-xx", "-f", "--pretty=none"])
+        .assert()
+        .stdout(indoc::formatdoc! {r#"
+            HTTP/1.1 200 OK
+            Date: N/A
+            Content-Type: text/plain
+            Content-Length: 5
+
+            key={c}
+        "#, c = "1"});
+}
+
+#[test]
+fn dont_compress_request_body_if_content_encoding_have_value() {
+    let server = server();
+    get_command()
+        .arg(format!("{}/normal", server.base_url()))
+        .args([
+            &format!("key={}", "1".repeat(1000)),
+            "content-encoding:gzip",
+            "-xx",
+            "-f",
+            "--pretty=none",
+        ])
+        .assert()
+        .stdout(indoc::formatdoc! {r#"
+            HTTP/1.1 200 OK
+            Date: N/A
+            Content-Type: text/plain
+            Content-Length: 1004
+
+            key={c}
+        "#, c = "1".repeat(1000),});
+}
+
+#[test]
+fn compress_body_from_file() {
+    let server = server::http(|req| async move {
+        assert_eq!("Hello world\n", zlib_decode(req.body().await).unwrap());
+        hyper::Response::default()
+    });
+
+    let dir = tempfile::tempdir().unwrap();
+    let filename = dir.path().join("input.txt");
+    OpenOptions::new()
+        .create(true)
+        .truncate(true)
+        .write(true)
+        .open(&filename)
+        .unwrap()
+        .write_all(b"Hello world\n")
+        .unwrap();
+
+    get_command()
+        .arg(server.base_url())
+        .arg("-xx")
+        .arg(format!("@{}", filename.to_string_lossy()))
+        .assert()
+        .success();
+}
+
+#[test]
+fn compress_body_from_file_unless_compress_rate_less_1() {
+    let server = server::http(|req| async move {
+        assert_eq!("Hello world\n", req.body_as_string().await);
+        hyper::Response::default()
+    });
+
+    let dir = tempfile::tempdir().unwrap();
+    let filename = dir.path().join("input.txt");
+    OpenOptions::new()
+        .create(true)
+        .truncate(true)
+        .write(true)
+        .open(&filename)
+        .unwrap()
+        .write_all(b"Hello world\n")
+        .unwrap();
+
+    get_command()
+        .arg(server.base_url())
+        .arg("-x")
+        .arg(format!("@{}", filename.to_string_lossy()))
+        .assert()
+        .success();
+}
+#[test]
+fn test_cannot_combine_compress_with_multipart() {
```

Comment:
Should we handle the situation where multipart requests, specified by headers and body, are currently unprocessed? 
Reading the request header Content-Type before compression should handle these cases.

Reply:
Gave it a try, but it seems I can't construct a multipart request on my own at the moment. won't worry about this anymore.

```
xh post https://postman-echo.com/post  --raw='--16e4fcaf49bf49a5b0d13aeeb3b8f817
Content-Disposition: form-data; name="a"

1
--16e4fcaf49bf49a5b0d13aeeb3b8f817--
'  content-type:"multipart/form-data; boundary=16e4fcaf49bf49a5b0d13aeeb3b8f817"
```